### PR TITLE
clash-verge-rev: build from source

### PR DESCRIPTION
**Pull request by oluceps**
clash-verge-rev: 1.6.6 -> 1.7.2-unstable-2024-07-04
Not tested by me Yet

@Guanran928 for test.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by linsui** [comment]
https://github.com/NixOS/nixpkgs/pull/318525

File: pkgs/by-name/cl/clash-verge-rev/package.nix [hunk 1]
```diff
@@ -1,19 +1,229 @@
...
+    mkdir -p ./resources
+    ln -s ${set_dns} resources/set_dns.sh
+    ln -s ${unset_dns} resources/unset_dns.sh
+    ln -s ${country} resources/Country.mmdb
+    ln -s ${v2ray-domain-list-community}/share/v2ray/geosite.dat resources/geosite.dat
+    ln -s ${v2ray-geoip}/share/v2ray/geoip.dat resources/geoip.dat
+
+    ln -s ${service_components}/bin/clash-verge-service resources/clash-verge-service
+    ln -s ${service_components}/bin/install-service resources/install-service
+    ln -s ${service_components}/bin/uninstall-service resources/uninstall-service
```
Comment (Guanran928) [inline pkgs/by-name/cl/clash-verge-rev/package.nix:170]: Does this work under NixOS?

File: pkgs/by-name/cl/clash-verge-rev/package.nix [hunk 2]
```diff
@@ -1,19 +1,229 @@
...
+
+  postPatch = ''
+    substituteInPlace ./tauri.conf.json \
+      --replace-fail '"distDir": "../dist",' '"distDir": "${ui}",' \
+      --replace-fail '"beforeBuildCommand": "pnpm run web:build"' '"beforeBuildCommand": ""'
+
+    # install mihomo
+    mkdir -p ./sidecar
+    ln -s ${mihomo}/bin/mihomo sidecar/verge-mihomo-${rust.envVars.rustHostPlatform}
+    ln -s ${mihomo-alpha}/bin/mihomo sidecar/verge-mihomo-alpha-${rust.envVars.rustHostPlatform}
```
Comment (Guanran928) [inline pkgs/by-name/cl/clash-verge-rev/package.nix:158]: I think they are now called `$out/bin/verge-mihomo` and `$out/bin/verge-mihomo-alpha`.

File: pkgs/by-name/cl/clash-verge-rev/package.nix (hunk 1 above)
Comment (oluceps) [reply pkgs/by-name/cl/clash-verge-rev/package.nix:170]: > Does this work under NixOS?

Not tested yet but I guess nope.

File: pkgs/by-name/cl/clash-verge-rev/package.nix [hunk 3]
```diff
@@ -1,19 +1,229 @@
...
+    hash = "sha256-3L31/3ODy2I5pRDVjKH46tVjTQ+2pJFKEkK+ZxBNr4E=";
+  };
+  unset_dns = fetchurl {
+    url = "https://github.com/clash-verge-rev/set-dns-script/releases/download/script/unset_dns.sh";
+    hash = "sha256-Nj0bYqUvtxxMVi1Kv5OydWrmO+fhJyMUkSeO8AnjWmo=";
+  };
+  country = fetchurl {
+    url = "https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/country.mmdb";
+    hash = "sha256-eb5jeWzpP/nblMBGTMDsXSYh/rD5jxk/RWpDNFtyhTs=";
+  };
```
[the remark this answers is not in the document]
Comment (oluceps) [reply pkgs/by-name/cl/clash-verge-rev/package.nix:65]: > Do they contain stable hashes?

Seems ain't but seems there's no reproducible ways to get the dns set scripts.

**Pull request closed without merging**